### PR TITLE
Follower adds entries to DB via non-libevent thread.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: cpp/fetcher/continuous_fetcher.cc [hunk 1]
```diff
@@ -123,23 +125,18 @@ void ContinuousFetcherImpl::FetchDone(Task* task) {
...
 
   if (restart_fetch_) {
-    base_->Add(bind(&ContinuousFetcherImpl::FetchDelayDone, this, nullptr));
+    executor_->Add(bind(&ContinuousFetcherImpl::FetchDelayDone, this));
   } else {
-    base_->Delay(seconds(FLAGS_delay_between_fetches_seconds),
-                 new Task(bind(&ContinuousFetcherImpl::FetchDelayDone, this,
-                               _1),
-                          base_));
+    executor_->Add([this]() {
```
Comment (pphaneuf) [inline cpp/fetcher/continuous_fetcher.cc]: I'd rather keep ```base_``` for the delay, this kind of sleeping in a thread pool callback is the kind of thing that's getting us in trouble... :wink:

**Comment by pphaneuf** [comment]
I sat down at my computer meaning to do *this*, only to find your PR. :smiley:

File: cpp/fetcher/continuous_fetcher.cc [hunk 2]
```diff
@@ -107,7 +109,7 @@ void ContinuousFetcherImpl::StartFetch(const unique_lock<mutex>& lock) {
   }
 
   fetch_task_.reset(
-      new Task(bind(&ContinuousFetcherImpl::FetchDone, this, _1), base_));
+      new Task(bind(&ContinuousFetcherImpl::FetchDone, this, _1), executor_));
```
Comment (pphaneuf) [inline cpp/fetcher/continuous_fetcher.cc]: The executor for this task is the one that is used to do the disk writes, no need to change anything else.

Comment (AlCutter) [inline cpp/fetcher/continuous_fetcher.cc]: Sorry, not thinking, done.

Comment (AlCutter) [inline cpp/fetcher/continuous_fetcher.cc]: oke doke.

File: cpp/fetcher/continuous_fetcher.cc (hunk 1 above)
Comment (AlCutter) [inline cpp/fetcher/continuous_fetcher.cc]: Sorry, not thinking, done.

File: cpp/fetcher/continuous_fetcher.cc [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 #include <gflags/gflags.h>
 #include <glog/logging.h>
+#include <thread>
```
Comment (pphaneuf) [inline cpp/fetcher/continuous_fetcher.cc]: I think this header is a leftover?

**Comment by pphaneuf** [comment]
LGTM, other than the leftover header. Thanks!

File: cpp/fetcher/continuous_fetcher.cc (hunk 3 above)
Comment (AlCutter) [inline cpp/fetcher/continuous_fetcher.cc]: dinged.

**Pull request merged**